src/test/java/org/aksw/iguana/cc/config/elements/ConnectionConfigTest.java

```diff
@@ -0,0 +1,107 @@
+                                "endpoint01",
+                                "0.1",
+                                new DatasetConfig("MyData", "some.ttl"),
+                                URI.create("http://example.com/sparql"),
+                                null, null, null),
+                        """
+                        {"name":"endpoint01","endpoint":"http://example.com/sparql","version":"0.1","authentication": null,"updateEndpoint":null,"dataset":{"name":"MyData","file":"some.ttl"}, "updateAuthentication": null}
+                        """
+                ),
+                Arguments.of(new ConnectionConfig( // test default values
+                                "endpoint01",
+                                null,
+                                null,
+                                URI.create("http://example.com/sparql"),
+                                null, null, null),
+                        """
+                        {
+                            "name":"endpoint01",
+                            "endpoint":"http://example.com/sparql"
+                        }
+                        """
+                ),
+                Arguments.of(new ConnectionConfig( // test setting everything
+                                "endpoint01",
+                                "v2",
+                                new DatasetConfig("dataset1", "some.ttl"),
+                                URI.create("http://example.com/sparql"),
+                                new ConnectionConfig.Authentication("user", "pass"),
+                                URI.create("http://example.com/update"),
+                                new ConnectionConfig.Authentication("user_update", "pass_update")),
+                        """
+                        {
+                            "name":"endpoint01",
+                            "version": "v2",
+                            "endpoint":"http://example.com/sparql",
+                            "authentication": {
+                                "user": "user",
+                                "password": "pass"
+                            },
+                            "updateEndpoint": "http://example.com/update",
+                            "updateAuthentication": {
+                                "user": "user_update",
+                                "password": "pass_update"
+                            },
+                            "dataset": {
+                                "name": "dataset1",
+                                "file": "some.ttl"
+                            }
+                        }
+                        """
+                )
+        );
+
+    }
+
+    @ParameterizedTest
+    @MethodSource("testData")
+    @Disabled("Doesn't work with testing default values and we don't serialize the configurations, so it's not really needed.")
+    public void testSerialization(ConnectionConfig config, String expectedJson) throws Exception {
```

Comment:
It might make sense to generate configs for users but then probably better directly in yaml. What exactly is the problem with the generated serializations?

Reply:
The problem was, that it needed a different testing set. The current and only test set for the connection config has json values that leave out properties of the ConnectionConfig, because they have default values. If you create a new ConnectionConfig, it instantiates those values and the mapper writes them out. This leads to the failure of the assertions.